components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingChallengeQuestionsHandlerTest.java

```diff
@@ -0,0 +1,396 @@
+import org.wso2.carbon.identity.application.common.model.IdentityProviderProperty;
+import org.wso2.carbon.identity.mgt.util.Utils;
+import org.wso2.carbon.identity.recovery.ChallengeQuestionManager;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+import org.wso2.carbon.identity.recovery.model.ChallengeQuestion;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Vector;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * This class will test the Post Authentication Handler for Missing Challenge Questions
+ **/
+@PrepareForTest({IdentityProviderManager.class, MultitenantUtils.class, Utils.class, FrameworkServiceDataHolder
+        .class, ChallengeQuestionManager.class, HttpServletResponse.class, ConfigurationFacade.class})
+public class PostAuthnMissingChallengeQuestionsHandlerTest {
+
+
+    private static final String CHALLENGE_QUESTIONS_REQUESTED = "challengeQuestionsRequested";
+
+    @Mock
+    private HttpServletRequest httpServletRequest;
+
+    @Mock
+    private HttpServletResponse httpServletResponse;
+
+    @Mock
+    private IdentityProviderManager identityProviderManager;
+
+    @Mock
+    private FrameworkServiceDataHolder frameworkServiceDataHolder;
+
+    @Mock
+    private ChallengeQuestionManager challengeQuestionManager;
+
+    @Mock
+    private ConfigurationFacade configurationFacade;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeMethod
+    public void setup() {
+        // initialize all the @Mock objects
+        MockitoAnnotations.initMocks(this);
+        // mock all the statics
+        mockStatic(HttpServletResponse.class);
+        mockStatic(IdentityProviderManager.class);
+        mockStatic(MultitenantUtils.class);
+        mockStatic(Utils.class);
+        mockStatic(FrameworkServiceDataHolder.class);
+        mockStatic(ChallengeQuestionManager.class);
+        mockStatic(ConfigurationFacade.class);
+    }
+
+    @Test
```

Comment:
Please add the annotation description and describe what you are trying to test? @Test(description=""